### DISCUSSION TITLE
Pure Python av wheel for python 3.11

### DISCUSSION THREAD

**Discussion by ranjitsingha** [PyAV-Org/PyAV#1684] [1. Help] [answered]
**IMPORTANT:** 
How do i generate pure python wheel for python 3.11 instead of cpython

Is it possible to generate pure Python wheel for av for python 3.11. If so then how.

The wheel contains files like:
```
utils.cpython-311.so
stream.cpython-311.so
logging.cpython-311.so
etc..
```
I want pure python .so

I have ffmpeg 7.1 installed and i am in termux and I am building av wheel from the github source code for arm64-v8a.
Using:
```
git clone https://github.com/PyAV-Org/PyAV.git
cd PyAV
make
make test
python setup.py bdist_wheel
```

The cpython wheel file installs and works absolutely fine on termux. 

But since i want to use av within Chaqopy i need pure python wheel.

**Comment by WyattBlue** [comment]
I need to clarify something important here - there seems to be a misunderstanding about ".so" files and "pure Python" wheels.

A "pure Python" wheel would not contain any .so (shared object) files at all. The .so files you're seeing are compiled C/C++ extensions, which is exactly what PyAV relies on since it's a wrapper around FFmpeg's C libraries. There's no such thing as a "pure Python .so" file - .so files are by definition compiled native code.

PyAV cannot be made into a pure Python package because its core functionality requires direct interaction with FFmpeg's C libraries. The package is inherently a C extension module.

**Reply by ranjitsingha** [reply]
> I need to clarify something important here - there seems to be a misunderstanding about ".so" files and "pure Python" wheels.
> 
> A "pure Python" wheel would not contain any .so (shared object) files at all. The .so files you're seeing are compiled C/C++ extensions, which is exactly what PyAV relies on since it's a wrapper around FFmpeg's C libraries. There's no such thing as a "pure Python .so" file - .so files are by definition compiled native code.
> 
> PyAV cannot be made into a pure Python package because its core functionality requires direct interaction with FFmpeg's C libraries. The package is inherently a C extension module.

But this tool can generate pure python wheel?https://github.com/pypa/cibuildwheel

**Reply by WyattBlue** [reply]
No, cibuildwheel cannot convert C extension modules into pure Python code. cibuildwheel is a tool that helps automate building wheels for different platforms and Python versions - it doesn't change the fundamental nature of the package being built.

**Comment by WyattBlue** [comment, marked as answer]
Look at https://github.com/chaquo/chaquopy/issues/302
